lib/saluki-components/src/sources/checks/mod.rs

```diff
@@ -0,0 +1,712 @@
+                match submit_runnable_check_req.send(runnable_check_request).await {
+                    Ok(_) => {
+                        trace!("Check request submitted to dispatcher");
+                    }
+                    Err(e) => {
+                        error!("Error submitting check request to dispatcher: {}", e);
+                    }
+                }
+            }
+            Some(deleted_entity) = deleted_entities.recv() => {
+                info!("Deleting check request for {}", deleted_entity.name);
+                match submit_stop_check_req.send(deleted_entity).await {
+                    Ok(_) => {
+                        debug!("Successfully sent stop check request");
+                    }
+                    Err(e) => {
+                        error!("Error sending stop check request: {}", e);
+                    }
+                }
+            }
+        }
+    }
+
+    debug!("Shutting down stream coordinator");
+    stream_shutdown_coordinator.shutdown().await;
+
+    info!("Check listener stopped.");
+
+    Ok(())
+}
+
+trait CheckScheduler {
+    fn can_run_check(&self, check_request: &RunnableCheckRequest) -> bool;
+    fn run_check(&mut self, check_request: &RunnableCheckRequest) -> Result<(), GenericError>;
+    fn stop_check(&mut self, check_name: CheckRequest);
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum PyMetricType {
+    Gauge = 0,
+    Rate,
+    Count,
+    MonotonicCount,
+    Counter,
+    Histogram,
+    Historate,
+}
+
+impl From<i32> for PyMetricType {
+    fn from(v: i32) -> Self {
+        match v {
+            0 => PyMetricType::Gauge,
+            1 => PyMetricType::Rate,
+            2 => PyMetricType::Count,
+            3 => PyMetricType::MonotonicCount,
+            4 => PyMetricType::Counter,
+            5 => PyMetricType::Histogram,
+            6 => PyMetricType::Historate,
+            _ => {
+                warn!("Unknown metric type: {}, considering it as a gauge", v);
+                PyMetricType::Gauge
+            }
+        }
+    }
+}
+
+#[derive(Debug, Snafu)]
+#[snafu(context(suffix(false)))]
+pub enum AggregatorError {
+    UnsupportedType {},
+}
+
+/// CheckMetric are used to transmit metrics from python check execution results
+/// to forward in the saluki's pipeline.
+#[derive(Debug, PartialEq)]
+pub struct CheckMetric {
+    name: String,
+    metric_type: PyMetricType,
+    value: f64,
+    tags: Vec<String>,
+}
+
+impl TryInto<Event> for CheckMetric {
+    type Error = AggregatorError;
+
+    fn try_into(self) -> Result<Event, Self::Error> {
+        // Convert Vec<String> to Vec<Tag>
+        let tags: Vec<Tag> = self.tags.into_iter().map(Tag::from).collect();
+
+        // Convert Vec<Tag> to TagSet
+        let tagset: TagSet = TagSet::from_iter(tags);
+
+        let context = Context::from_parts(self.name, tagset);
+        let metadata = MetricMetadata::default();
+
+        match self.metric_type {
```

Comment:
This is where the "magic" from the `try_into` happens, we just ignore any metric type that isn't gauge/counter.